pudb/debugger.py

```diff
@@ -342,6 +342,22 @@ def set_frame_index(self, index):
 
         self.ui.stack_list._w.set_focus(self.ui.translate_ui_stack_index(index))
 
+    def open_file_to_edit(self, index):
+        self.curframe, line_number = self.stack[index]
+        filename = self.curframe.f_code.co_filename
+
+        editor = os.environ.get("EDITOR", "nano")
+        editor = editor.lower()
+
+        editor_command = f"{editor}"
+        if editor in {"vi", "vim", "nvim", "emacs"}:
+            editor_command = f"{editor} +{line_number}"
+        elif editor == "nano":
+            editor_command = f"{editor} -l {line_number}"
```

Comment:
This fails if an explicit path is specified, or if the command is `emacs-gtk` or some-such. Maybe check for substrings? It might also be useful to let the user customize the invocation in settings.